core/services/wifi/Hotspot.py

```diff
@@ -0,0 +1,339 @@
+                    exit_code = self._subprocess.returncode
+                    raise RuntimeError(f"Failed to initialize Hostapd ({exit_code}).")
+            if not self._dhcp_server:
+                self._dhcp_server = DHCPServerManager(self._ap_interface_name, self._ipv4_gateway)
+                return
+            self._dhcp_server.restart()
+        except Exception as error:
+            raise RuntimeError(f"Unable to start hotspot. {error}") from error
+
+    def stop(self) -> None:
+        logger.info("Stopping hotspot.")
+        if self.is_running():
+            assert self._subprocess is not None
+            self._subprocess.kill()
+            if not self._dhcp_server:
+                logger.warning("Cannot stop DHCP server for hotspot, as was already not running.")
+                return
+            self._dhcp_server.stop()
+        else:
+            logger.info("Tried to stop hostpot, but it was already not running.")
+
+    def restart(self) -> None:
+        self.stop()
+        self.start()
+
+    def is_running(self) -> bool:
+        return self._subprocess is not None and self._subprocess.poll() is None
+
+    @staticmethod
+    def config_path() -> pathlib.Path:
+        config_dir = pathlib.Path(tempfile.tempdir or "/")
+        return config_dir.joinpath("hostapd.conf")
+
+    def hostapd_config(self) -> str:
+        desired_channel_frequency = self.desired_channel_frequency()
+        desired_channel_number = HotspotManager.channel_number(desired_channel_frequency)
+
+        return (
+            "# WiFi interface to be used (in this case a virtual one)\n"
+            f"interface={self._ap_interface_name}\n"
+            "# Channel (frequency) of the access point\n"
+            f"channel={desired_channel_number}\n"
+            "# SSID broadcasted by the access point\n"
+            f'ssid2="{self._ap_ssid}"\n'
+            "# Passphrase for the access point\n"
+            f"wpa_passphrase={self._ap_passphrase}\n"
+            "# Use the 2.4GHz band\n"
+            f"hw_mode={HostapdFrequency.mode_from_channel_frequency(desired_channel_frequency).value}\n"
+            "# Accept all MAC addresses\n"
+            "macaddr_acl=0\n"
+            "# Use WPA authentication\n"
+            "auth_algs=1\n"
+            "# Require clients to know the network name\n"
+            "ignore_broadcast_ssid=0\n"
+            "# Use WPA2\n"
+            "wpa=2\n"
+            "# Use a pre-shared key\n"
+            "wpa_key_mgmt=WPA-PSK\n"
+            "wpa_pairwise=TKIP\n"
+            "rsn_pairwise=CCMP\n"
+        )
+
+    def _create_temp_config_file(self) -> None:
+        logger.info(f"Saving temporary hostapd config file on {self.config_path()}")
+        with open(self.config_path(), "w", encoding="utf-8") as f:
+            f.write(self.hostapd_config())
+
+    @staticmethod
+    def device_serial() -> str:
+        try:
+            with open("/proc/cpuinfo", "r", encoding="utf-8") as f:
```

Comment:
Since the device serial information is not something that will exist in different boards or computers under cpuinfo, it's better to rely on: `/etc/machine-id`.
https://www.freedesktop.org/software/systemd/man/machine-id.html

Reply:
I'll leave here an interesting article about ids: http://0pointer.de/blog/projects/ids.html